connector-j-8/src/main/java/com/google/cloud/sql/mysql/SocketFactory.java

```diff
@@ -102,4 +98,16 @@ public Socket performTlsHandshake(SocketConnection socketConnection, ServerSessi
 
   @Override
   public void afterHandshake() {}
+
+  // Returns True if running in a Google App Engine Standard runtime, otherwise False.
+  private boolean runningOnGaeStandard(){
```

Comment:
Is there anywhere common you could put this method, maybe as a static method, instead of copying it three times?

Reply:
The only place that I'm aware of would be in the core module. We could add a new class to hold helpers, or add it in the SslSocketFactory (but I'm not sure it makes sense there). What are your thoughts?

Reply:
Do these modules already depend on the core module? If so, that sounds like the right place. I think it makes sense to add it to a new class. I usually prefer class names that describe the class rather than "helper". This is about the runtime environment, right, so how about a class called `Environment`?

Reply:
Yes, all 4 depend on the core module. I think it would be best to handle this in a separate PR, and perhaps consider moving more of the guts of the connect methods to a shared class as well. I'll add it as an action item to the doc.